<launch>
  <!-- Vehicle namespace -->
  <arg name="uuv_name" />
  <!-- Flag, if true, read disturbance specifications from a file -->
  <arg name="use_file" default="false" />
  <!-- Filename -->
  <arg name="disturbance_file" default=""/>
  <!-- In case the file is not used, then the user can set one current
    perturbation and one body wrench through the launch file -->

  <!-- Current perturbation -->
  <!-- The current is applied as an absolute setting -->
  <arg name="current_starting_time" default="20.0"/>
  <arg name="current_vel" default="1"/>
  <arg name="current_horz_angle" default="0.0"/>
  <arg name="current_vert_angle" default="0.0"/>
  <arg name="current_duration" default="10"/>
  <!-- Link wrench -->
  <!-- Forces and torques are summed up -->
  <arg name="force_x" default="0"/>
  <arg name="force_y" default="2000"/>
  <arg name="force_z" default="0"/>
  <arg name="torque_x" default="0"/>
  <arg name="torque_y" default="0"/>
  <arg name="torque_z" default="0"/>
  <arg name="wrench_starting_time" default="30"/>
  <arg name="wrench_duration" default="20"/>

  <group>
    <push-ros-namespace namespace="$(var uuv_name)"/>
    <group if="$(var use_file)">
        <node pkg="uuv_control_utils"
          exec="disturbance_manager.py"
          name="disturbance_manager"
          output="screen">
          <!-- Load  parameters -->
          <param from="$(var disturbance_file)"/>
        </node>
    </group>

    <group unless="$(var use_file)">
    <node pkg="uuv_control_utils"
        exec="disturbance_manager.py"
        name="disturbance_manager"
        output="screen">
      <param name="disturbances">
        <param name="d1">
          <param name="type" value="current"/>
          <param name="starting_time" value="$(var current_starting_time)"/>
          <param name="velocity" value="$(var current_vel)"/>
          <param name="horizontal_angle" value="$(var current_horz_angle)"/>
          <param name="vertical_angle" value="$(var current_vert_angle)"/>
          <param name="duration" value="$(var current_duration)"/>
        </param>
        <param name="d2">
          <param name="type" value="wrench"/>
          <param name="starting_time" value="$(var wrench_starting_time)"/>
          <param name="duration" value="$(var wrench_duration)"/>
          <param name="force" value="$(var force_x), $(var force_y), $(var force_z)" value-sep=", "/>
          <param name="torque" value="$(var torque_x), $(var torque_y), $(var torque_z)" value-sep=", "/>
        </param>
      </param>
    </node>
  </group>
</group>

</launch>
